File: packages/go/logger/type.go
package logger

import (
	"github.com/rudderlabs/rudder-observability-kit/packages/go/labels"
)

type Logger interface {
	Debug(string, ...labels.Label)
	Info(string, ...labels.Label)
	Warn(string, ...labels.Label)
	Error(string, ...labels.Label)
	Fatal(string, ...labels.Label)
	GetLabels() []labels.Label
	CreateLogger(...labels.Label) Logger
}

File: packages/python/logger/type.py
from typing import Self
from ..labels.type import Label
class Logger:
    def __init__(self, *labels: Label) -> None:
        self.labels: list[Label] = list(labels)
    def get_labels(self) -> list[Label]:
        return self.labels
    def create_logger(self, *labels: Label) -> Self:
        allLabels = self.get_labels() + list(labels)
        return Logger(*allLabels)
    
    # Child classes should override these methods
    def debug(self, msg: str, *labels: Label):
        pass
    def info(self, msg: str, *labels: Label):
        pass
    def warn(self, msg: str, *labels: Label):
        pass
    def error(self, msg: str, *labels: Label):
        pass
    def fatal(self, msg: str, *labels: Label):
        pass
